RELEASES.md

```diff
@@ -197,11 +199,107 @@ To create a minor release:
+     which can change during patch releases, and only field additions are valid patch changes.)
 4. Open a PR with these changes that will land them back on `v0.35.x`
 5. Once this change has landed on the backport branch, make sure to pull it locally, then push a tag.
    - `git tag -a v0.35.1 -m 'Release v0.35.1'`
    - `git push origin v0.35.1`
 6. Create a pull request back to master with the CHANGELOG & version changes from the latest release.
-   - Remove all `R:minor` labels from the pull requests that were included in the release.
+   - Remove all `R:patch` labels from the pull requests that were included in the release.
    - Do not merge the backport branch into master.
+
+## Minor Release Checklist
+
+The following set of steps are performed on all releases that increment the
+_minor_ version, e.g. v0.25 to v0.26. These steps ensure that Tendermint is
+well tested, stable, and suitable for adoption by the various diverse projects
+that rely on Tendermint.
+
+### Feature Freeze
+
+Ahead of any minor version release of Tendermint, the software enters 'Feature
+Freeze' for at least two weeks. A feature freeze means that _no_ new features
+are added to the code being prepared for release. This period directly follows
+the creation of the [backport branch](#creating-a-backport-branch). The
+Tendermint team instead directs all attention to ensuring that the existing
+code is stable and reliable. Broken tests are fixed, flakey-tests are remedied,
+end-to-end test failures are thoroughly diagnosed and all efforts of the team
+are aimed at improving the quality of the code. During this period, the upgrade
+harness tests are run repeatedly and a variety of in-house testnets are run to
+ensure Tendermint functions at the scale it will be used by application
+developers and node operators.
+
+### Nightly End-To-End Tests
+
+The Tendermint team maintains [a set of end-to-end
+tests](https://github.com/tendermint/tendermint/blob/master/test/e2e/README.md#L1)
+that run each night on the latest commit of the project and on the code in the
+tip of each backport branch. These tests start a network of containerized
+Tendermint processes and run automated checks that the network functions as
+expected in both stable and unstable conditions. During the feature freeze,
+these tests are run nightly and must pass consistently for a release of
+Tendermint to be considered stable.
+
+### Upgrade Harness
+
+The Tendermint team is creating an upgrade test harness to exercise the
+workflow of stopping an instance of Tendermint running one version of the
+software and starting up the same application running the next version. To
+support upgrade testing, we will add the ability to terminate the Tendermint
+process at specific pre-defined points in its execution so that we can verify
+upgrades work in a representative sample of stop conditions.
+
+### Large Scale Testnets
```

Comment:
One thing that hasn't been covered in this section is transactional load profiles. The e2e testnet only blows gently on the network. Alongside the aforementioned p2p scenarios we should also benchmark these large testnets under considerable load. It may simply be about firing transactions at a single node where the mempool has to be able to shed it or we could also expand it to more unique load situations including:
- Submitting to one or many different nodes
- Experimenting with bursts vs constant load
- Certain transaction properties (large txs, small txs, invalid txs)

Where the e2e testnet should focus on correctness, these large scale testnets are about focusing on performance aspects

Reply:
> Where the e2e testnet should focus on correctness, these large scale testnets are about focusing on performance aspects.

I definitely agree with this.

>Submitting to one or many different nodes
Experimenting with bursts vs constant load
Certain transaction properties (large txs, small txs, invalid txs)

I think that these could be more readily conceived of as benchmarks that are written as part of the unit test suite for the code. Go provides a nice set of tools for writing [function benchmarks](https://pkg.go.dev/testing#hdr-Benchmarks) that should be leaned on as much as possible for this kind of testing. I think we should focus these tests on things that are much more specific to large-scale deployments. The question of "How well does the mempool deal with a transaction load of profile XYZ" can easily be answered by writing a benchmark that generates many large or small tx and fires them quickly or in a bursty way. I'm not really sure we need to spin up a multi-hundred node network to understand that.

I'm also not certain this needs to happen with each release, but should happen when we make specific changes to the mempool itself.

Reply:
I think we should start with consistent load. The more profiles etc. that we test for, the more work we need to do to design and shape them in our tools and the more analysis we'll need to perform to understand what exactly we're seeing.